core/src/test/java/com/coderanch/blackjack/HandTest.java

```diff
@@ -0,0 +1,349 @@
+     *
+     * @param scoreTest contains the test hand, and the target score.
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void bestScore_hasCorrectScore(BestScoreTestArgument scoreTest) {
+        var hand = createHand(scoreTest.cards);
+        assertThat(
+            "The best score must match the target",
+            hand.bestScore(),
+            is(scoreTest.getTargetScore())
+        );
+    }
+
+    /**
+     * Tests that {@link Hand#withAdditionalCard(Card)} maintains the correct score.
+     *
+     * @param hand the hand to test.
+     * @param card the card to add to the hand.
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void withAdditionalCard_hasCorrectScore(Hand hand, Card card) {
+        if (card.rank() == Rank.ACE) {
+            return;
+        }
+        var newHand = hand.withAdditionalCard(card);
+        assertThat(
+            "The difference must be correct",
+            newHand.bestScore() - hand.bestScore(),
+            is(equalTo(card.rank().points()))
+        );
+    }
+
+    /**
+     * Tests that passing {@code null} for the first card
+     * when constructing a new hand causes an exception to be thrown.
+     *
+     * @param card the card to construct the hand with.
+     */
+    @Theory(nullsAccepted = false)
+    @SuppressWarnings("checkstyle:methodname")
+    public void newHand_withNullFirstCard_throwsException(Card card) {
+        assertThrows("Hand must throw an exception.", IllegalArgumentException.class, () -> {
+            new Hand(null, card);
+        });
+    }
+
+    /**
+     * Tests that passing {@code null} for the second card
+     * when constructing a new hand causes an exception to be thrown.
+     *
+     * @param card the card to construct the hand with.
+     */
+    @Theory(nullsAccepted = false)
+    @SuppressWarnings("checkstyle:methodname")
+    public void newHand_withNullSecondCard_throwsException(Card card) {
+        assertThrows("Hand must throw an exception.", IllegalArgumentException.class, () -> {
+            new Hand(card, null);
+        });
+    }
+
+    /**
+     * Tests that passing {@code null} for {@code card} when adding a new card causes an exception to be thrown.
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void addCard_withNullCard_throwsException() {
+        var hand = new Hand(new Card(Rank.ACE, Suit.CLUBS), new Card(Rank.ACE, Suit.CLUBS));
+        assertThrows("Hand must throw an exception.", IllegalArgumentException.class, () -> {
+            hand.withAdditionalCard(null);
+        });
+    }
+
+    /**
+     * Tests that hands that {@link Hand#isBust()} returns the correct answer.
+     *
+     * @param bustHand the hand to check if bust.
+     */
+    @Theory
+    public void isBust(BustHand bustHand) {
+        assertThat(
+            "The hand must be bust.",
+            bustHand.hand.isBust(),
+            is(equalTo(true))
```

Comment:
`equalTo` is mostly used to make the matcher read more fluently. It doesn't actually do anything. So in cases like this, you can just write `is(true)`.